satpy/tests/reader_tests/test_viirs_gran.py

```diff
@@ -0,0 +1,242 @@
+                                                   dims=('Rows', 'Columns'))
+            file_content['Smoke_Mask'].attrs['units'] = 1
+            file_content['Smoke_Mask'].attrs['_FillValue'] = -128
+            file_content['Smoke_Mask'].attrs['valid_range'] = [0, 3]
+            file_content['Smoke_Mask/shape'] = (768, 3200)
+        else:
+            print('filetype_info is: ', filetype_info)
+            print('filetype is: ', filetype_info['file_type'])
+            assert False
+
+        return file_content
+
+
+class TestVIIRSGRANReader(unittest.TestCase):
+    """
+    Test VIIRS GRAN reader
+    """
+    yaml_file = 'viirs_gran.yaml'
+
+    def setUp(self):
+        ''' Wrap the netCDF4 file handler with the fake handler '''
+        from satpy.config import config_search_paths
+        from satpy.readers.viirs_gran import VIIRSGRANFileHandler
+        self.reader_configs = config_search_paths(
+            os.path.join('readers', self.yaml_file))
+        self.p = mock.patch.object(
+            VIIRSGRANFileHandler, '__bases__', (FakeNetCDF4FileHandler3,))
+        self.fake_handler = self.p.start()
+        self.p.is_local = True
+
+    def tearDown(self):
+        ''' Unwrap the netCDF4 file handler '''
+        self.p.stop()
+
+    def test_init(self):
+        ''' Test basic initialization of the reader '''
+        from satpy.readers import load_reader
+        r = load_reader(self.reader_configs)
+        loadables = r.select_files_from_pathnames([
+            'JRR-CloudMask_v2r0_npp_s201903060621591'
+            '_e201903060623233_c201903060804490.nc',
+            ])
+        self.assertTrue(len(loadables), 1)
+        r.create_filehandlers(loadables)
+        self.assertTrue(r.file_handlers)
+
+    def test_load_available_datasets(self):
+        """ Test loading datasets """
+        from satpy.readers import load_reader
+        from satpy import DatasetID
+        r = load_reader(self.reader_configs)
+        loadables = r.select_files_from_pathnames([
+            'JRR-CloudMask_v2r0_npp_s201903060621591'
+            '_e201903060623233_c201903060804490.nc',
+            ])
+        r.create_filehandlers(loadables)
+        lat_shape = None
+        for var_name, val in self.file_content.items():
+            if var_name == 'Latitude':
+                lat_shape = self[var_name + "/shape"]
+                break
+        handled_variables = set()
+
+        for is_avail, ds_info in (configured_datasets or []):
+            if is_avail is not None:
+                yield is_avail, ds_info
+            var_name = ds_info.get('file_key', ds_info['name'])
+            matches = self.file_type_matches(ds_info['file_type'])
+            if matches and var_name in self:
+                handled_variables.add(var_name)
+                new_info = ds_info.copy()
+                yield True, new_info
+            elif is_avail is None:
+                yield is_avail, ds_info
```

Comment:
This looks like you've copied some of your logic from the file handler. Since you have these `yield` statements this function is treated as a generator and never actually executed. Could you take a look and fix this up.